wezterm-ssh/Cargo.toml

```diff
@@ -39,6 +39,7 @@ wezterm-uds = { path = "../wezterm-uds" }
 
 # Not used directly, but is used to centralize the openssl vendor feature selection
 async_ossl = { path = "../async_ossl" }
+scopeguard = "1.2.0"
```

Comment:
while scopeguard is handy in a pinch, I'm not in favor of it here for a couple of reasons:

* It is an additional external dependency
* You still have to remember to initiate the scope guard, and that is required in multiple places in this PR

What I'd prefer to see as a solution for this is for `connect_to_host` to return a little wrapper around the Child:

```rust
struct KillOnDropChild(Child);

impl std::ops::Deref for KillOnDropChild { ... }
impl std::ops::DerefMut for KillOnDropChild { ... }
impl Drop for KillOnDropChild {
   fn drop(&mut self) {
       self.0.kill();
       self.0.wait();
   }
}
```